src/node_sea.cc

```diff
@@ -248,7 +273,42 @@ ExitCode GenerateSingleExecutableBlob(const SeaConfig& config) {
     return ExitCode::kGenericUserError;
   }
 
-  SeaResource sea{config.flags, main_script};
+  std::vector<char> snapshot_blob;
+  bool builds_snapshot_from_main =
+      static_cast<bool>(config.flags & SeaFlags::kUseSnapshot);
+  if (builds_snapshot_from_main) {
+    SnapshotData snapshot;
+    std::vector<std::string> patched_args = {args[0], GetAnonymousMainPath()};
```

Comment:
Should we use the executable filename / the original script name (the code_path variable being added in https://github.com/nodejs/node/pull/48191) instead for the main script file name?